### PR TITLE
Fix/restrict worker node instance profile

### PR DESCRIPTION
**Pull request by naris-silpakit**
### What does this PR do?

Restricts access to instance profiles in worker node groups by:
 - disabling imds access for self-managed node groups by updating the default metadata_options in the launch-templates module:
 ```
    http_endpoint               = "enabled"
    http_tokens                 = "required"
    http_put_response_hop_limit = 1
```
 - disabling imds access for managed node groups by updating the default context in the root module:
```
    http_endpoint               = "enabled"
    http_tokens                 = "required"
    http_put_response_hop_limit = 1
```

Previously there was a comment in the code saying "Hop limit should be between 2 and 64 for IMDSv2 instance metadata services", but that is likely outdated as the [guidance here](https://aws.github.io/aws-eks-best-practices/security/docs/iam/#restrict-access-to-the-instance-profile-assigned-to-the-worker-node) recommends setting the hop limit to 1.

This PR also fixes a typo in the managed node group example. The x86 bottlerocket managed node group entry was named "bottlerocket_arm". I renamed this to "bottlerocket_x86".

### Motivation

Restricting access to the instance profiles in worker node groups minimizes the blast radius of a breach as described in the best practices guide above.

### More

- [x] Yes, I have tested the PR using my local account setup  (Provide any test evidence report under Additional Notes)
- [ ] Yes, I have added a new example under [examples](https://github.com/aws-samples/aws-eks-accelerator-for-terraform/tree/main/examples) to support my PR
- [ ] Yes, I have created another PR for add-ons under [add-ons](https://github.com/aws-samples/ssp-eks-add-ons) repo (if applicable)
- [ ] Yes, I have updated the [docs](https://github.com/aws-samples/aws-eks-accelerator-for-terraform/tree/main/docs) for this feature
- [x] Yes, I ran `pre-commit run -a` with this PR

**Note**: Not all the PRs required examples and docs except a new pattern or add-on added.

### For Moderators
- [ ] E2E Test successfully complete before merge?

### Additional Notes

Testing notes:
   - Deployed a managed node group and a self-managed node group, accessed their respective EC2 instances and tried a request to IMDSv1 using `curl -v http://169.254.169.254/latest/meta-data/` and confirmed I received an "Unauthorized" response.

### CODE REVIEW

**Review by vara-bonthu — CHANGES_REQUESTED** [review]
@naris-silpakit What problem that this PR is solving?

File: locals.tf [hunk 1]
```diff
@@ -12,7 +12,7 @@ locals {
     # http details
     http_endpoint               = "enabled"
     http_tokens                 = "required"
-    http_put_response_hop_limit = 2 # Hop limit should be between 2 and 64 for IMDSv2 instance metadata services
+    http_put_response_hop_limit = 1
```
[the remark this answers is not in the document]
Comment (bryantbiggs) [reply locals.tf:15]: I believe the ALB ingress controller requires a hop  limit of at least 2 as well as kube2iam, and probably others

[the remark this answers is not in the document]
Comment (naris-silpakit) [reply locals.tf:15]: This guidance recommends setting the hop limit to 1: https://aws.github.io/aws-eks-best-practices/security/docs/iam/#restrict-access-to-the-instance-profile-assigned-to-the-worker-node

This changes the instances to be more secure by default. One option we could take is to update the launch-templates for anything that requires a hop limit >2 and not use the default configuration. Alternatively if we want to leave the hop-limit as 2 I can close this PR.

[the remark this answers is not in the document]
Comment (bryantbiggs) [reply locals.tf:15]: just came across this as well:
> If any containers that you deploy to the node group use the Instance Metadata Service Version 2, make sure to set the Metadata response hop limit to 2 in your launch template. For more information, see Instance metadata and user data in the Amazon EC2 User Guide for Linux Instances. If you deploy a managed node group without using a custom launch template, this value is automatically set for the node group in the default launch template.

https://docs.aws.amazon.com/eks/latest/userguide/launch-templates.html

**Comment by naris-silpakit** [comment]
Discussed with @askulkarni2 and we agree that this PR isn't needed anymore. I'm going to close it.

**Pull request closed without merging**